src/webgpu/api/validation/query_set/create.spec.ts

```diff
@@ -0,0 +1,141 @@
+export const description = `
+Tests for validation in createQuerySet.
+`;
+
+import { poptions } from '../../../../common/framework/params_builder.js';
+import { makeTestGroup } from '../../../../common/framework/test_group.js';
+import { ValidationTest } from '../validation_test.js';
+
+export const g = makeTestGroup(ValidationTest);
+
+const kMaxQueryCount = 8192;
+
+// Test with these query counts.
+const kQueryCounts: number[] = [
+  // valid
+  0,
+  kMaxQueryCount,
+  // invalid
+  kMaxQueryCount + 1,
+];
+
+g.test('occlusion')
+  .desc(
+    `
+Tests that create query set with the type of occlusion.
+- control cases:
+  - count <= kMaxQueryCount
+  - pipelineStatistics is undefined or empty
+- invalid cases:
+  - count > kMaxQueryCount
+  - pipelineStatistics is set with pipeline statistic names
+  `
+  )
+  .params([
+    ...poptions('count', kQueryCounts),
+    ...poptions('pipelineStatistics', [
+      // valid
+      undefined,
+      [],
+      // invalid
+      ['clipper-invocations'],
+    ]),
```

Comment:
You can use `as const` here and then the type cast on `t.params` should no longer be needed.
```suggestion
    ] as const),
```